akka-http2-support/src/main/scala/akka/http/impl/engine/http2/PriorKnowledgeSwitch.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import javax.net.ssl.SSLException
+
+import akka.util.ByteString
+import akka.NotUsed
+import akka.annotation.InternalApi
+import akka.http.impl.engine.server.HttpAttributes
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.stream.TLSProtocol.{ SessionBytes, SessionTruncated, SslTlsInbound, SslTlsOutbound }
+import akka.stream.scaladsl.Flow
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import akka.stream._
+
+/** INTERNAL API */
+@InternalApi
+private[http] object PriorKnowledgeSwitch {
```

Comment:
Seems pretty verbose for what it does, but I can't really think of a more straightforward way either.

Reply:
I'd be more than happy to cut down the noise here, open to all suggestions!

Reply:
@jrudolph I thought about that, but then again, the AlpnSwitch deals with Bidi and this works with Flow, so I think the correct move in that case would be to abstract each separately so that you'd be reusable for other stages. That does seem orthogonal to this PR though?